### DISCUSSION TITLE
[Feature Request] Slider - custom step values

### DISCUSSION THREAD

**Discussion by SinnerK0N** [heroui-inc/heroui#4737] [Feature requests]
### Is your feature request related to a problem? Please describe.

In some cases it doesnt make sense to give the user the choice to pick any value on a slider. For example when using a slider to filter through the sizes of a hard drives, valid values could be 128, 256, 512 GB so it doesnt make sense to let the user pick a value like 333 GB because it doesnt exist, it should instead jump to the nearest specified value so in this case it could be either 256 GB or 512 GB.

### Describe the solution you'd like

It would be nice to have a feature where you could set valid steps and the slider will allow the user to select from those values only.
Here is a working example of it in MUI: https://mui.com/material-ui/react-slider/#restricted-values

### Describe alternatives you've considered

I thought of using checkboxes but it can be overwhelming when there are too many values and its annoying when you want to select a range of values because you have to select all of the values in the range manually.

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1255/[feature-request]-slider-custom-step-values">ENG-1255 [Feature Request] Slider - custom step values</a></p>

**Comment by sidyr6002** [comment]
I would like to work on this feature. Can this be assigned to me?

**Comment by SinnerK0N** [comment]
bump
